### PR TITLE
feat(sdk): Add account data methods to Account

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

File: crates/matrix-sdk/src/account.rs [hunk 1]
```diff
@@ -623,4 +628,34 @@ impl Account {
         });
         Ok(self.client.send(request, None).await?)
     }
+
+    /// Set the given account data event.
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:632]: The docs here a bit, how should i put it, spartanic. As it stands it's a bit hard to figure out what `content` can be.

If I take a look at the docs for [GlobalAccountDataEventContent], I'm greeted with a blanket trait implementation. Following the docs to [EventContent] shows me that absolutely everything implements `EventContent`. Following the docs to [GlobalAccountDataEventType] isn't clear right away either.


[GlobalAccountDataEventContent]: https://docs.ruma.io/ruma/events/trait.GlobalAccountDataEventContent.html
[EventContent]: https://docs.ruma.io/ruma/events/trait.EventContent.html
[GlobalAccountDataEventType]: https://docs.ruma.io/ruma/events/enum.GlobalAccountDataEventType.html

One has to note that `GlobalAccountDataEventType` is an associated type for the `EventType` trait, then filter through the implementors of `EventType` which use that associated type.

Can we improve those docs a bit and/or add an example?

File: crates/matrix-sdk/src/account.rs [hunk 2]
```diff
@@ -623,4 +628,34 @@ impl Account {
...
+    {
+        let own_user =
+            self.client.user_id().ok_or_else(|| Error::from(HttpError::AuthenticationRequired))?;
+
+        let request = set_global_account_data::v3::Request::new(&content, own_user)?;
+
+        Ok(self.client.send(request, None).await?)
+    }
+
+    /// Set the given raw account data event.
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:648]: Same  here, a bit more verbosity could help and an example wouldn't hurt either.

**Review by poljar — CHANGES_REQUESTED** [review]

File: crates/matrix-sdk/src/account.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk/src/account.rs:632]: Hm, I guess we could actually have a proper list of impls for all individual event content traits. Good point that it's currently pretty hard to find the matching types.

File: crates/matrix-sdk/src/account.rs (hunk 2 above)
Comment (jplatte) [reply crates/matrix-sdk/src/account.rs:648]: Heh, I was kinda expecting this PR not to go through w/o more docs. Maybe I should have put it up as draft only 😅

Comment (poljar) [reply crates/matrix-sdk/src/account.rs:648]: Ah right, you didn't request review. Sorry :sweat_smile:

Comment (jplatte) [reply crates/matrix-sdk/src/account.rs:648]: I did, actually. Personally I wouldn't need more docs and it was kinda late yesterday, I'll do better next time!

Comment (poljar) [reply crates/matrix-sdk/src/account.rs:648]: Yeah, obviously I don't need the docs either, but the docs aren't for us in the first place.

File: crates/matrix-sdk/src/account.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk/src/account.rs:632]: I've created https://github.com/ruma/ruma/issues/1268 for that.

**Comment by jplatte** [comment]
I wonder, should we also update the store from `set_account_data[_raw]`?

**Comment by jplatte** [comment]
I will update these methods to only deal with event content types, as suggested by pixlwave in #982.

**Comment by jplatte** [comment]
Done, looking for another review now.

**Review by poljar — APPROVED** [review]
Looks good.

**Pull request merged**